siphon/catalog.py

```diff
@@ -41,11 +41,23 @@ def __init__(self, catalog_url):
         # top level server url
         self.catalog_url = catalog_url
         self.base_tds_url = catalog_url.split('/thredds/')[0]
+
+        session = create_http_session()
+
         # get catalog.xml file
-        xml_fobj = urlopen(catalog_url)
+        resp = session.get(self.catalog_url)
+
+        # If we were given an HTML link, warn about it and try to fix to xml
+        if 'html' in resp.headers['content-type']:
+            import warnings
```

Comment:
The logic that was suggested before was: logging for warnings of bad conditions that the user may want to know about, but can't do anything about; warnings module for warnings the user should address. 

https://docs.python.org/3/howto/logging.html

Specifically: "warnings.warn() in library code if the issue is avoidable and the client application should be modified to eliminate the warning"